ddtrace/tracer/textmap.go

```diff
@@ -264,46 +265,71 @@ func (p *chainedPropagator) Inject(spanCtx ddtrace.SpanContext, carrier interfac
 // based on the precedence order of the propagators. Generally, the first valid
-// trace context that could be extracted will be returned, and other extractors will
-// be ignored. However, the W3C tracestate header value will always be extracted and
-// stored in the local trace context even if a previous propagator has already succeeded
-// so long as the trace-ids match.
+// trace context that could be extracted will be returned. However, the W3C tracestate
+// header value will always be extracted and stored in the local trace context even if
+// a previous propagator has already succeeded so long as the trace-ids match.
+// Furthermore, if we have already successfully extracted a trace context and a
+// subsequent trace context has conflicting trace information, such information will
+// be relayed in the original trace context with a SpanLink.
 func (p *chainedPropagator) Extract(carrier interface{}) (ddtrace.SpanContext, error) {
-	var ctx ddtrace.SpanContext
+	var ctx ddtrace.SpanContext // First valid trace context will be stored here
+	var links []ddtrace.SpanLink
 	for _, v := range p.extractors {
-		if ctx != nil {
-			// A local trace context has already been extracted.
-			pw3c, isW3C := v.(*propagatorW3c)
-			if !isW3C {
-				continue // Ignore other propagators.
+		extractedCtx, err := v.Extract(carrier)
+		// Handling extraction errors. If it is the first extraction, return error. Else, ignore
+		if err != nil {
+			if ctx == nil && err != ErrSpanContextNotFound {
+				return nil, err
+			} else {
+				continue
 			}
-			w3cCtx, err := pw3c.Extract(carrier)
-			if err == nil && w3cCtx.(*spanContext).TraceID128() == ctx.(*spanContext).TraceID128() {
-				pw3c.propagateTracestate(ctx.(*spanContext), w3cCtx.(*spanContext))
-				if w3cCtx.SpanID() != ctx.SpanID() {
-					var ddCtx *spanContext
-					if ddp := getDatadogPropagator(p); ddp != nil {
-						if ddSpanCtx, err := ddp.Extract(carrier); err == nil {
-							ddCtx, _ = ddSpanCtx.(*spanContext)
+		}
+		if ctx != nil { // A local trace context has already been extracted
+			propagatorType, err := getPropagatorType(v)
+			if err != nil { // Propagator is not one of the known, supported types. Ignore this extraction
+				log.Debug("Propagator type is unknown. Skipping extraction")
+				continue
+			}
+			if extractedCtx.(ddtrace.SpanContextW3C).TraceID128() == ctx.(ddtrace.SpanContextW3C).TraceID128() {
+				if propagatorType == "tracecontext" {
+					v.(*propagatorW3c).propagateTracestate(ctx.(*spanContext), extractedCtx.(*spanContext)) //Sending tracestate to the ctx to be returned
+					if extractedCtx.SpanID() != ctx.SpanID() {
+						var ddCtx *spanContext
+						if ddp := getDatadogPropagator(p); ddp != nil {
+							if ddSpanCtx, err := ddp.Extract(carrier); err == nil {
+								ddCtx, _ = ddSpanCtx.(*spanContext)
+							}
 						}
+						overrideDatadogParentID(ctx.(*spanContext), extractedCtx.(*spanContext), ddCtx)
+					}
+				}
+			} else {
```

Comment:
```suggestion
			} 
			// handle non-w3c propagators
			else {
```

Reply:
Syntactically, `else` must be on the same line as the closing `if` bracket. I added the comment inline